src/app/content/highlights/hooks/utils.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  GetHighlightsColorsEnum,
+  GetHighlightsSourceTypeEnum,
+  Highlight,
+  Highlights
+} from '@openstax/highlighter/dist/api';
+import omit from 'lodash/fp/omit';
+import { AppServices, AppState } from '../../../types';
+import { assertDefined } from '../../../utils';
+import { book as bookSelector } from '../../selectors';
+import { Book } from '../../types';
+import { stripIdVersion } from '../../utils';
+import * as select from '../selectors';
+import { HighlightLocationFilters, SummaryHighlightsPagination, } from '../types';
+import { addSummaryHighlight, getHighlightLocationFilterForPage } from '../utils';
+import { getNextPageSources } from '../utils/paginationUtils';
+
+export const formatReceivedHighlights = (
+  highlights: Highlight[],
+  locationFilters: HighlightLocationFilters
+) => highlights.reduce((result, highlight) => {
+  const pageId = stripIdVersion(highlight.sourceId);
+  const location = assertDefined(
+    getHighlightLocationFilterForPage(locationFilters, pageId),
+    'highlight is not in a valid location'
+  );
+  const locationFilterId = stripIdVersion(location.id);
+
+  return addSummaryHighlight(result, {
+    highlight,
+    locationFilterId,
+    pageId,
+  });
+}, {} as ReturnType<typeof addSummaryHighlight>);
+
+export const incrementPage = (pagination: NonNullable<SummaryHighlightsPagination>) =>
+  ({...pagination, page: pagination.page + 1});
+
+export const getNewSources = (state: AppState, omitSources: string[], pageSize?: number) => {
+  const book = bookSelector(state);
+  const remainingCounts = omit(omitSources, select.filteredCountsPerPage(state));
+  return book ? getNextPageSources(remainingCounts, book.tree, pageSize) : [];
+};
+
+export const extractDataFromHighlightClientResponse = (highlightsResponse: Highlights) => {
```

Comment:
can you update `src/app/content/highlights/hooks/locationChange.ts` to use this function to reduce duplication?